app/org/sagebionetworks/bridge/play/controllers/ReportController.java

```diff
@@ -0,0 +1,142 @@
+package org.sagebionetworks.bridge.play.controllers;
+
+import static org.sagebionetworks.bridge.Roles.DEVELOPER;
+import static org.sagebionetworks.bridge.Roles.WORKER;
+
+import org.apache.commons.lang3.StringUtils;
+import org.joda.time.LocalDate;
+import org.springframework.beans.factory.annotation.Autowired;
+
+import org.sagebionetworks.bridge.exceptions.BadRequestException;
+import org.sagebionetworks.bridge.json.DateUtils;
+import org.sagebionetworks.bridge.models.DateRangeResourceList;
+import org.sagebionetworks.bridge.models.accounts.StudyParticipant;
+import org.sagebionetworks.bridge.models.accounts.UserSession;
+import org.sagebionetworks.bridge.models.reports.ReportData;
+import org.sagebionetworks.bridge.models.studies.Study;
+import org.sagebionetworks.bridge.services.ParticipantService;
+import org.sagebionetworks.bridge.services.ReportService;
+
+import com.fasterxml.jackson.databind.JsonNode;
+
+import play.mvc.Result;
+
+public class ReportController extends BaseController {
+
+    @Autowired
+    ReportService reportService;
+    
+    @Autowired
+    ParticipantService participantService;
+    
+    final void setReportService(ReportService reportService) {
+        this.reportService = reportService;
+    }
+    
+    final void setParticipantService(ParticipantService participantService) {
+        this.participantService = participantService;
+    }
+    
+    /**
+     * Reports for specific individuals accessible only to consented study participants.
+     */
+    public Result getParticipantReport(String identifier, String startDateString, String endDateString) {
+        UserSession session = getAuthenticatedAndConsentedSession();
```

Comment:
Reports built against this service need to pass the correct headers in, and that's a client side issue. That's a bug in the existing mPower 1.3 app. Here we expect consented users and that's correct. They must be consented.